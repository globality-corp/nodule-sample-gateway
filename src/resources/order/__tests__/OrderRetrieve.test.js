import request from 'supertest';
import { Nodule } from '@globality/nodule-config';
import { signSymmetric } from '@globality/nodule-graphql';
import { mockResponse } from '@globality/nodule-openapi';
import createApp from '../../../app';

let app;

const email = 'user@globality.com';
const token = signSymmetric({ email });


beforeEach(async () => {
    await Nodule.testing().fromObject(
        mockResponse('charmander', 'order.retrieve', {
            id: '00000000-0000-0000-0000-000000000000',
            customerId: '00000000-0000-0000-0000-111111111111',
        }),
    ).load();

    app = createApp();
});

it('should retrieve a single order', async () => {
    const query = `
        query retrieveOrder($orderId: ID) {
          order(id: $orderId) {
            items {
                id
                customerId
            }
          }
        }
    `;

    const variables = {
        orderId: '00000000-0000-0000-0000-111111111111',
    };


    const result = await request(app).post(
        '/gql/graphql',
    ).set(
        'Authorization', `Bearer ${token}`,
    ).send({
        query,
        variables,
    });

    expect(result.body.errors).toBeUndefined();
    expect(result.body).toMatchSnapshot();
});
