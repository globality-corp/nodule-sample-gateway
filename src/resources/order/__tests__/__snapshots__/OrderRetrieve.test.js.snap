// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should retrieve a single order 1`] = `
Object {
  "data": Object {
    "order": Object {
      "items": Array [
        Object {
          "customerId": "00000000-0000-0000-0000-111111111111",
          "id": "00000000-0000-0000-0000-000000000000",
        },
      ],
    },
  },
}
`;
